controller/embedding/manager.py

```diff
@@ -134,4 +109,13 @@ def get_embedding_name(
         raise ValueError("attribute not found")
     attribute_name = attribute_item.name
 
-    return f"{attribute_name}-{embedding_type}-{embedding_handle}"
+    name = f"{attribute_name}-{embedding_type}-{platform}"
+
+    if model:
+        name += f"-{model}"
+
+    if apiToken:
+        name += f"-{apiToken[:8]}..."
```

Comment:
haven't tested how big of an overlapp there is between key (so e.g. if the first 10 char always describe the organization and the rest something else this might have some issues).

I couldn't find a pattern or anything  listed somewhere but I found how openai displays them

![image](https://github.com/code-kern-ai/refinery-gateway/assets/91723236/a956c37d-31bd-450a-b405-66af9bc876c9)

so maybe `apiToken[:3]+"..."+apiToken[-4:]`

Reply:
solved